File: include/gui/GUIHandler.r
#ifndef _GUIHANDLER_R_
#define _GUIHANDLER_R_

#include "gui/GUIObj.r"

#define CONT_GUIHANDLER \
	CONT_GUIOBJ \
	void (*event) ( void *view, void *ev );

struct GUIHandler {
	CONT_GUIHANDLER
};

#endif

File: src/gui/GUIHandler.c
#include "gui/GUIHandler.r"
#include <assert.h>

#include "gui/GUIHandler.h"

void event ( void *_self, void *ev ) {
	struct GUIHandler **self = _self;

	assert ( self );
	if ( (*self)->event ) {
		(*self)->event ( _self, ev );
	}
}
